lib/Conversion/TorchOnnxToTorch/DefaultDomainQtoZ.cpp

```diff
@@ -758,6 +758,124 @@ void mlir::torch::onnx_c::populateDefaultDomainQtoZ(
+              // inferred whether the dim:1 is reduced or dim:2. To avoid these
+              // type of cases, the check has been placed.
+              SmallVector<int64_t> reduceDims;
+              unsigned resultShapeCounter = 0;
+              for (unsigned i = 0; i < inputShape.size(); i++) {
+                if (resultShapeCounter < resultShape.size() &&
+                    inputShape[i] == resultShape[resultShapeCounter]) {
+                  resultShapeCounter++;
+                } else {
+                  reduceDims.push_back(i);
+                  if (resultShapeCounter < resultShape.size() &&
+                      resultShape[resultShapeCounter] == 1)
+                    resultShapeCounter++;
+                }
+              }
+              for (auto i : reduceDims) {
+                axesList.push_back(rewriter.create<Torch::ConstantIntOp>(
+                    binder.getLoc(), rewriter.getI64IntegerAttr(i)));
+              }
+            }
+          }
+          if (axesList.empty()) {
+            Torch::BaseTensorType axesType =
+                axesVal.getType().cast<Torch::BaseTensorType>();
+            auto axesTy = dyn_cast<Torch::ValueTensorType>(axesVal.getType());
+            auto axesShape = axesTy.getSizes();
+            if (axesShape.size() != 1 || axesShape[0] == Torch::kUnknownSize)
+              return failure();
+
+            Value zero = rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getType<Torch::IntType>(),
+                rewriter.getI64IntegerAttr(0));
+            SmallVector<int64_t> selectSizes{1};
+            auto selType = rewriter.getType<Torch::ValueTensorType>(
+                selectSizes, axesType.getOptionalDtype());
+            int64_t numAxes = axesShape[0];
+            for (int64_t i = 0; i < numAxes; ++i) {
+              Value iv = rewriter.create<Torch::ConstantIntOp>(
+                  binder.getLoc(), rewriter.getType<Torch::IntType>(),
+                  rewriter.getI64IntegerAttr(i));
+              Value extract = rewriter.create<Torch::AtenSelectIntOp>(
+                  binder.getLoc(), selType, axesVal, zero, iv);
+              Value dim = rewriter.create<Torch::AtenItemOp>(
+                  binder.getLoc(), rewriter.getType<Torch::IntType>(), extract);
+              axesList.push_back(dim);
+            }
+          }
+        }
+
+        SmallVector<int64_t> axesInts;
+        if (!binder.s64IntegerArrayAttr(axesInts, "axes", {})) {
+          for (int64_t i = 0, s = axesInts.size(); i < s; ++i) {
+            Value iv = rewriter.create<Torch::ConstantIntOp>(
+                binder.getLoc(), rewriter.getType<Torch::IntType>(),
+                rewriter.getI64IntegerAttr(axesInts[i]));
+            axesList.push_back(iv);
+          }
+        }
+
+        if (axesList.empty() && noop_with_empty_axes) {
+          rewriter.replaceOp(binder.op, data);
```

Comment:
This should be changed to use "operand" instead of "data", since you do not want to include the absolute value op in the noop case.